src/pages/guides/index.md

```diff
@@ -286,22 +286,57 @@ You can also consider implementing a retry mechanism to call public key urls in
 
 **Verifying the Signature**
 
-Once you have the public keys fetched as plain text, you can now verify the digital signatures by following the steps as below:
+Once you have the pem public keys, you can now verify the digital signatures by following the steps as below:
 
-1. Decrypt the message digest using the public key.
-2. Compute the hash message digest of the event using `rsa-sha256` hash function algorithm. For computing:
-   - Use the webhook request payload for event deliveries.
-   - Use the challenge code in the query param for challenge deliveries.
-3. Validate each signature by comparing:
-   - the digest received (in step 1) after decrypting the signature using the public key.
-   - and the message digest computed (in step 2) by hashing.
-4. If any one of the signature validation is successful, then the event is valid.
+1. Create the PublicKey object using the pem public key.
+2. For Java based application
+   - create a Signature instance using the `SHA256withRSA` algorithm
+   - initialize the instance using the public key
+   - supply this instance the raw payload 
+   - use this instance to verify by passing the decoded signature
+3. For nodeJS based application
+   - create a crypto verifier using `RSA-SHA256` algorithm
+   - suppliy this verifier the raw payload buffer object
+   - pass the public key and the decoded signature to the verifier
+
+4. Do the above for both the signatures and if any one of the signature validations is successful, then the event is valid.
```

Comment:
I really don't think these steps are helping. We already have added the code for these steps now.
I would keep the previous text and highlight that event payload used for signature verification should be raw:

>- Use the **raw** webhook request payload for event deliveries.

For all the points that we have come across, we can instead add some pointers like:

- How to extract payload from an express app?
- Make sure that the payload is not going through any conversion.
- Maybe add the code snippets for Node10/12 as well.

Reply:
We can't keep the previous text as that was basically demonstrating the steps for signature verification using `Cipher` verification strategy. Go through [this](https://adobeio.slack.com/archives/C6841FMPU/p1647998397999849?thread_ts=1647972756.218329&cid=C6841FMPU) slack thread to understand more on this.
> I would keep the previous text

Reply:
The reason I mentioned these steps explicitly is that user couldn't relate to what steps were documented and the code was doing. Now they can. Still, I think only adding some extra explicit note about **no payload conversion** should suffice.

Adding code snippets for different versions seems too much IMHO. The bottomline is that, user should be implementing the steps referring to what is documented here and then customize their code based on the language and env he is using.

Reply:
>We can't keep the previous text as that was basically demonstrating the steps for signature verification using Cipher verification strategy. Go through [this](https://adobeio.slack.com/archives/C6841FMPU/p1647998397999849?thread_ts=1647972756.218329&cid=C6841FMPU) slack thread to understand more on this.

>The reason I mentioned these steps explicitly is that user couldn't relate to what steps were documented and the code was doing. Now they can.

Ok then let's change the text to make it generic and add comments in the code snippets so that the users can correlate.

Reply:
>Adding code snippets for different versions seems too much IMHO. The bottomline is that, user should be implementing the steps referring to what is documented here and then customize their code based on the language and env he is using.

I don't see any harm in supporting our customers and enriching our documentation. We already have the code for Node10 [here](https://git.corp.adobe.com/stanwar/sample-node-app). Our NodeJS SDK covers Node 12 and above, and now we will soon have the Java code snippet.
If you still think that it seems too much, then we can tackle this when we get another request for Node10, apart from the CCX team.

Reply:
sounds good